src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -318,6 +318,35 @@ class BookingService(
     return GetBookingForPremisesResult.Success(booking)
   }
 
+  @Transactional
+  fun deleteBooking(booking: BookingEntity) {
```

Comment:
I wondered if it was possible to factor away the extra knowledge this method has to have about all the possible booking associations. Doing so would reduce the repetition and therefore make the code easier to change. Any new association or change wouldn't have to be weaved through multiple places. 

[Rails leverages SQL to do this type of thing](https://apidock.com/rails/ActiveRecord/Base/destroy) for us in a DRY way:

```
# File activerecord/lib/active_record/base.rb, line 2614
      def destroy
        unless new_record?
          connection.delete(
            "DELETE FROM #{self.class.quoted_table_name} " +
            "WHERE #{connection.quote_column_name(self.class.primary_key)} = #{quoted_id}",
            "#{self.class.name} Destroy"
          )
        end

        @destroyed = true
        freeze
      end
```

I don't think we need to take action on this, just sharing some idle thoughts on how it might be even better incase it's of interest 🙂